pkg/kn/commands/config/channel_type_mapping.go

```diff
@@ -0,0 +1,89 @@
+// Copyright © 2022 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configCmd
+
+import (
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	"knative.dev/client/pkg/kn/commands"
+	"knative.dev/client/pkg/kn/commands/config/channel"
+	"knative.dev/client/pkg/kn/config"
+)
+
+var defaultChannelTypeUpdater = viperUpdater(viper.GetViper(), config.KeyChannelTypeMappings)
+
+// NewConfigChannelTypeMappingCommand represents 'kn config channel-type-mapping' command
+func NewConfigChannelTypeMappingCommand(p *commands.KnParams) *cobra.Command {
+	channelTypeMappingCmd := &cobra.Command{
+		Use:   "channel-type-mapping",
+		Short: "Manage channel type mapping list",
+	}
+	channelTypeMappingCmd.AddCommand(newChannelTypeDeleteCommand(p))
+	channelTypeMappingCmd.AddCommand(newChannelTypeAddCommand(p))
+	channelTypeMappingCmd.AddCommand(newChannelTypeUpdateCommand(p))
+	channelTypeMappingCmd.AddCommand(newChannelTypeListCommand(p))
+	return channelTypeMappingCmd
+}
+
+func addChannelTypeMappingFlags(flagSet *pflag.FlagSet, channelType *config.ChannelTypeMapping) {
+	flagSet.StringVar(&channelType.Kind, "kind", "", "The kind of Channel, such as InMemoryChannel or KafkaChannel, based on the default ConfigMap")
+	flagSet.StringVar(&channelType.Group, "group", "", "The API group of the Kubernetes resource")
+	flagSet.StringVar(&channelType.Version, "version", "", "The version of the Kubernetes resource")
+}
+
+func newChannelTypeUpdateCommand(_ *commands.KnParams) *cobra.Command {
+	channelType := config.ChannelTypeMapping{}
+	updateChannelTypeMapCmd := &cobra.Command{
+		Use:   "update",
+		Short: "Update channel type mapping by its alias",
+		Args:  cobra.ExactArgs(1),
+		RunE:  modifyList(channel.Update(&channelType), defaultChannelTypeUpdater),
+	}
+	addChannelTypeMappingFlags(updateChannelTypeMapCmd.Flags(), &channelType)
+	return updateChannelTypeMapCmd
+}
+
+func newChannelTypeAddCommand(_ *commands.KnParams) *cobra.Command {
+	channelType := config.ChannelTypeMapping{}
+	addChannelTypeMapCmd := &cobra.Command{
+		Use:   "add",
+		Short: "Add a new channel type mapping",
```

Comment:
In the same consistency sake. I'd like to get some opinions on `add` vs. `create`. I'd go with `create` for mappings.

@maximilien @rhuss